src/drivers/imx/sai.c

```diff
@@ -0,0 +1,312 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2019 NXP
+//
+// Author: Daniel Baluta <daniel.baluta@nxp.com>
+// Author: Jerome Laclavere <jerome.laclavere@nxp.com>
+// Author: Guido Roncarolo <guido.roncarolo@nxp.com>
+
+#include <sof/drivers/edma.h>
+#include <sof/audio/component.h>
+#include <sof/drivers/sai.h>
+#include <sof/lib/dai.h>
+#include <sof/lib/dma.h>
+#include <ipc/dai.h>
+
+#define REG_RX_DIR 0
+#define REG_TX_DIR 1
+
+static void sai_start(struct dai *dai, int direction)
+{
+	tracev_sai("SAI: sai_start");
+
+	direction = (direction == SOF_IPC_STREAM_PLAYBACK);
```

Comment:
I feel we are going backwards with this. You should use values directly and change your macros to assume rx = 1.

Reply:
ok

Reply:
@groncarolo Gotta remember that because in my ESAI driver I'm also guilty of this.

Reply:
:)